### ISSUE TITLE
copy behaviour of annotations and object identity

### ISSUE THREAD

**Issue by JuliaSprenger** [NeuralEnsemble/python-neo#811]
Annotations of objects are by default not deepcopied when manipulating an object, e.g. upon slicing of a signal. However, array annotations are explicitly deepcopied (see https://github.com/NeuralEnsemble/python-neo/issues/809#issuecomment-619902320), whereas this is not the case for regular annotations.
This can lead to unexpected behaviour, e.g. when identifying objects via annotations (see #809).
 
I am not sure at which point it would be best to fix this behaviour. There are multiple options here:

- extend the `__get_item__` methods in a similar fashion as implemented for `array_annotations`. This would lead to multiple steps in the process of slicing. First linking of `annotations` by `__array_finalize` and then explicit deepcopying later on.
- extend the `__array_finalize__` methods as these are called also during the process of slicing
- introduce unique object identifies within neo. This would require an extended discussion first about the concept of identifiers to use and their expected behaviour in Neo.

@apdavison @samuelgarcia Do you have a preference regarding the different options here?

**Comment by samuelgarcia** [comment]
Hi Julia.
I think really prefer deepcopy of annotation. 
I forget why we did not copy annotations yet.

Having uid for neo object is not a solution I like.

And about the when to make the copy, I don't known choose the simpler!!

Cheers,

Samuel

**Comment by samuelgarcia** [comment]
I think this is important to put in 0.9 no ? Even with something very simple.
This is the last stuff. No ?

**Comment by JuliaSprenger** [comment]
I don't think this will go into the next release as there is not enough time to implement and review it properly. @samuelgarcia @apdavison I suggest to postpone this to 0.11.0. even though it is labeled as blocker.
